### DISCUSSION TITLE
List number of batches for each local rank

### DISCUSSION THREAD

**Discussion by obrookes** [Lightning-AI/pytorch-lightning#13430] [General]
Hi everyone, 

Is there an existing mechanism to print the number of batches i.e. len(dataloader) being processed by each rank in a multi-gpu setting?

Thanks :)

**Comment by akihironitta** [comment]
`trainer.estimated_stepping_batches` may be what you're looking for.
https://pytorch-lightning.readthedocs.io/en/1.6.4/api/pytorch_lightning.trainer.trainer.Trainer.html#pytorch_lightning.trainer.trainer.Trainer.estimated_stepping_batches

**Reply by obrookes** [reply]
Thanks for this @akihironitta!

**Comment by rohitgr7** [comment]
in a multi-GPU setting with DDP all the ranks processes same number of batches.

**Reply by rohitgr7** [reply]
`trainer.estimated_stepping_batches` gives you total optimization steps. for batches you can just do `len(self.train_dataloader)`

**Reply by obrookes** [reply]
Thanks for your reply @rohitgr7, it's much appreciated :+1:. As I understand it, samples are replicated to ensure each rank processes the same number of batches - is there a mechanism to display / control which classes are up-sampled? I'm working on a long-tailed recognition problem where this could potentially have a large impact.
